instrumentation/opentelemetry_ex_aws/lib/opentelemetry_ex_aws/dynamo_db.ex

```diff
@@ -0,0 +1,180 @@
+        %{attribute: Conventions.aws_dynamodb_total_segments(), path: "$.TotalSegments"},
+        %{attribute: Conventions.aws_dynamodb_attributes_to_get(), path: "$.AttributesToGet"},
+        %{attribute: Conventions.aws_dynamodb_consistent_read(), path: "$.ConsistentRead"},
+        %{attribute: Conventions.aws_dynamodb_index_name(), path: "$.IndexName"},
+        %{attribute: Conventions.aws_dynamodb_limit(), path: "$.Limit"},
+        %{attribute: Conventions.aws_dynamodb_projection(), path: "$.ProjectionExpression"},
+        %{attribute: Conventions.aws_dynamodb_select(), path: "$.Select"},
+        %{attribute: Conventions.aws_dynamodb_table_names(), path: "$.TableName", process: &List.wrap/1}
+      ],
+      response: [
+        %{attribute: Conventions.aws_dynamodb_count(), path: "$.Count"},
+        %{attribute: Conventions.aws_dynamodb_scanned_count(), path: "$.ScannedCount"},
+        %{attribute: Conventions.aws_dynamodb_consumed_capacity(), path: "$.ConsumedCapacity"}
+      ]
+    },
+    "UpdateItem" => %{
+      request: [
+        %{attribute: Conventions.aws_dynamodb_table_names(), path: "$.TableName", process: &List.wrap/1}
+      ],
+      response: [
+        %{attribute: Conventions.aws_dynamodb_consumed_capacity(), path: "$.ConsumedCapacity"},
+        %{attribute: Conventions.aws_dynamodb_item_collection_metrics(), path: "$.ItemCollectionMetrics"}
+      ]
+    },
+    "UpdateTable" => %{
+      request: [
+        %{attribute: Conventions.aws_dynamodb_attribute_definitions(), path: "$.AttributeDefinitions"},
+        %{attribute: Conventions.aws_dynamodb_global_secondary_index_updates(), path: "$.GlobalIndexes"},
+        %{attribute: Conventions.aws_dynamodb_provisioned_read_capacity(), path: "$.ReadCapacity"},
+        %{attribute: Conventions.aws_dynamodb_provisioned_write_capacity(), path: "$.WriteCapacity"},
+        %{attribute: Conventions.aws_dynamodb_table_names(), path: "$.TableName", process: &List.wrap/1}
+      ],
+      response: [
+        %{attribute: Conventions.aws_dynamodb_consumed_capacity(), path: "$.ConsumedCapacity"}
+      ]
+    }
+  }
+
+  def get_attributes(%{operation: "DynamoDB" <> _rest} = metadata) do
+    [_service, method] = String.split(metadata.operation, ".", parts: 2)
+
+    attribute_definition = Map.get(@attributes, method, %{})
+
+    request_attributes(metadata, attribute_definition)
+    |> Map.merge(response_attributes(metadata, attribute_definition))
+    |> Map.put(Conventions.db_system(), "dynamodb")
+  end
+
+  defp request_attributes(%{request_body: request_body}, attribute_definition) do
+    extract_attributes(request_body, Map.get(attribute_definition, :request, []))
+  end
+
+  defp response_attributes(%{response_body: response_body}, attribute_definition) do
+    extract_attributes(response_body, Map.get(attribute_definition, :response, []))
+  end
+
+  defp response_attributes(%{result: :error}, _), do: %{}
```

Comment:
shouldn't there be a `Tracer.set_status()` call based on the error result?
As an example in `req`:
https://github.com/open-telemetry/opentelemetry-erlang-contrib/blob/main/instrumentation/opentelemetry_req/lib/opentelemetry_req.ex#L108

Reply:
I think it depends. An error response doesn't mean what the Span is doing errored, just that the response from the backend was an error, if that makes sense...

Reply:
Maybe we should set error, 

The tesla middleware assumes the span is an error if the response is 4xx.